configuration_go/abstr/kubernetes/thanos/compactor/compactor.go

```diff
@@ -1 +1,375 @@
 package compactor
+
+import (
+	"fmt"
+	"maps"
+	"net"
+	"time"
+
+	cmdopt "github.com/observatorium/observatorium/configuration_go/abstr/kubernetes/cmdoption"
+	"github.com/observatorium/observatorium/configuration_go/k8sutil"
+	k8sutilv2 "github.com/observatorium/observatorium/configuration_go/k8sutil/v2"
+	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	dataVolumeName      string = "data"
+	defaultHTTPPort     int    = 10902
+	defaultNamespace    string = "observatorium"
+	servicePortName     string = "http"
+	thanosContainerName string = "thanos"
+)
+
+// CompactorOptions represents the options/flags for the compactor.
+// See https://thanos.io/tip/components/compact.md/#flags for details.
+type CompactorOptions struct {
```

Comment:
Yes this is probably feasible with additional work. I'll think about it. 
But even today, it would be great if Thanos exported all these flags so that we would simply import them. @saswatamcode wanted to do that, but does not seem straightforward at all with current Thanos flags implementation.
With `cmdopts` we still have to maintain the flags list ourselves. Seems reasonable to me.